gigascience/gigadb-website
# Make cert check specific to the deploy env

# Pull request for issue: #1057 

This is a pull request for the following functionalities:

Perform open ssl cert check during  stage `build_staging` or `build_live`,  so the deployment of each env would be independent from the expired cert of other env.

## How to test?
Check the `build_staging` or `build_live` log in gitlab pipeline, you would only see staging url and `Certificate expires in: * days` in `build_staging` log, live url in `build_live` log.

## How have functionalities been implemented?

Move cert check from stage `integration_tests` to stage `build_staging/live`, so only the server with valid open ssl cert could be built. Moreover, the cert check for staging or live servers is performed separately.

## Any issues with implementation?
None

## Any changes to automated tests?
None

## Any changes to documentation?
None

## Any technical debt repayment?
None

## Any improvements to CI/CD pipeline?

Extract `cert_check` from `integration_tests` and put it to `gigadb-build-jobs.yml`,  this could make `build_staging` or `build_live` independently.
